core/src/App.svelte

```diff
@@ -366,14 +333,8 @@
                 showLoadingIndicator = false;
                 clearTimeout(loadingIndicatorTimeout);
               }
-            } else if (prop === 'tabNav') {
-              tabNav = obj.tabNav;
-            } else if (prop === 'isNavigateBack') {
-              isNavigateBack = obj.isNavigateBack;
-            } else if (prop === 'goBackContext') {
-              goBackContext = obj.goBackContext;
-            } else if (prop === 'isNavigationSyncEnabled') {
-              isNavigationSyncEnabled = obj.isNavigationSyncEnabled;
+            } else {
+              eval(prop + " = " + JSON.stringify(obj[prop]));
```

Comment:
This is a very interesting solution. Definitely cleaner. However, we want refrain from using the **eval** function altogether as it introduces issues with Content Security Policies. To allow this properly we need to enable [**'unsafe-eval'**](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy#unsafe-eval)

We had major issues with this before in the past trying to remove all direct and indirect usage of **eval** https://github.com/SAP/luigi/issues/1458